DEV-INTRO.md

```diff
@@ -1,59 +1,76 @@
-PEcAn Development
-=================
+# PEcAn Development
 
-Directory Structure
--------------------
+This is a minimal guide to getting started with PEcAn development.
```

Comment:
I think it would help if we put a note here (e.g. "how to follow this document:") to tell people to read all doc first, and follow the steps we itemize at the end, like a true quickstart, e.g.:
```
1. git clone git@github.com/pecanproject/pecan
2. Changed directory to be in the pecan directory, cd pecan
3. Have docker-compose.override.yml and .env in the pecan folder
4. Change this in docker-compose.override.yml
5. Change that in .env
6. start database: docker-compose -p pecan up -d postgres
7. ...
```
 What I struggled before with our current documentation was that I wasn't sure where it wants me to take action (i.e. start typing the code) versus when it just tries to explain something to me. So I would suggest explicitly dividing this doc into first reading, then doing.

Reply:
I agree with @ister, I think that would be a very useful way to structure this quickstart.

Reply:
now if only we got accepted in season of docs :(

Reply:
OK, let's not have this hold this PR, once it is in, I can test it on my laptop and make this addition